src/ldp/operations/SimplePutOperationHandler.ts

```diff
@@ -0,0 +1,35 @@
+import { Operation } from './Operation';
+import { OperationHandler } from './OperationHandler';
+import { ResourceStore } from '../../storage/ResourceStore';
+import { ResponseDescription } from './ResponseDescription';
+import { UnsupportedHttpError } from '../../util/errors/UnsupportedHttpError';
+
+/**
+ * Handles PUT {@link Operation}s.
+ * Calls the setRepresentation function from a {@link ResourceStore}.
+ */
+export class SimplePutOperationHandler extends OperationHandler {
+  private readonly store: ResourceStore;
+
+  public constructor(store: ResourceStore) {
+    super();
+    this.store = store;
+  }
+
+  public async canHandle(input: Operation): Promise<void> {
+    if (input.method !== 'PUT') {
+      throw new UnsupportedHttpError('This handler only supports PUT operations.');
+    }
+    if (!input.body) {
+      throw new UnsupportedHttpError('PUT operations require a body.');
+    }
+  }
+
+  public async handle(input: Operation): Promise<ResponseDescription> {
+    if (!input.body) {
+      throw new UnsupportedHttpError('PUT operations require a body.');
+    }
```

Comment:
`canHandle` has this already?

Reply:
I had a comment on this in the PR when changing to `strict`, but there were no comments on it so I didn't change it anymore afterwards. 

> There are some cases where a check that was in the `canHandle` function was needed again in the `handle` function to prevent accessing potential `undefined` data since some input fields are optional. E.g. the method field in a `HttpRequest`. Another option would be to leave behind that duplicate check and tell TypeScript to trust us that the data is there (since it should have been checked in `canHandle`). Not that big of an issue but might become a problem in the future if more advanced checks are needed.

So could also get around it with the `!` operator if we assume the `canHandle` function is always called first.